### PR TITLE
Add java-doc for implemented methods of io.netty.util.concurrent.Future#cancel(boolean mayInterruptIfRunning) 

### LINKED ISSUE #7609: should add java-doc for method DefaultPromise#cancel(boolean mayInterruptIfRunning)
### Expected behavior
implementation of method `io.netty.util.concurrent.DefaultPromise#cancel(boolean mayInterruptIfRunning)` actually ignored the param `mayInterruptIfRunning`.
so it need to make a comment like following:
(this comment is copied from `java.util.concurrent.CompletableFuture#cancel`):

>      * @param mayInterruptIfRunning this value has no effect in this
>      * implementation because interrupts are not used to control
>      * processing.
the other method `io.netty.util.concurrent.CompleteFuture#cancel` may do the same too.
### Actual behavior
there is no comment about the param `mayInterruptIfRunning`, which will make user confusing when call the `cancel` method with a param `mayInterruptIfRunning`.
### Netty version
4

### PR DESCRIPTION
**Pull request by laosijikaichele**
Motivation:

The methods implement `io.netty.util.concurrent.Future#cancel(boolean mayInterruptIfRunning)` which actually ignored the param mayInterruptIfRunning.We need to add comments for the `mayInterruptIfRunning` param.

Modifications:

Add comments for the `mayInterruptIfRunning` param.

Result:

People who call the `cancel` method will be more clear about the effect of `mayInterruptIfRunning` param.
Fixes #7609

### CODE REVIEW

File: common/src/main/java/io/netty/util/concurrent/CompleteFuture.java [hunk 1]
```diff
@@ -140,6 +140,9 @@ public boolean isCancelled() {
         return false;
     }
 
+    /**
+     * @param mayInterruptIfRunning this value has no effect in this implementation.
```
Comment (Scottmitch) [inline common/src/main/java/io/netty/util/concurrent/CompleteFuture.java]: at a minimum we need {@inheritDoc}, and in general I'm not convinced this change is necessary as the `cancel` operation itself is indicated as best effort in the original javadocs and therefore the parameter may not be used in general.

Comment (laosijikaichele) [reply common/src/main/java/io/netty/util/concurrent/CompleteFuture.java]: if a param value does not used in the method, then it should be indicated by the javadoc, but the original javadoc from `java.util.concurrent.Future#cancel` does not clearly indicate it, so the JVM team itself indicate it when the implementation method does not use the param `mayInterruptIfRunning`, please refer the javadoc of `java.util.concurrent.CompletableFuture#cancel` and `java.util.concurrent.ForkJoinTask#cancel` ;
 which javadocs clearly indicate that `mayInterruptIfRunning` has no effect in the implementation.

File: transport/src/main/java/io/netty/channel/DelegatingChannelPromiseNotifier.java [hunk 2]
```diff
@@ -172,6 +172,11 @@ public Void getNow() {
         return delegate.getNow();
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @param mayInterruptIfRunning this value has no effect in this implementation.
```
Comment (normanmaurer) [inline transport/src/main/java/io/netty/channel/DelegatingChannelPromiseNotifier.java]: @laosijikaichele actually I think this is not correct as it just delegates it and if it is ignored or not depends on the implementation of `delegate`

Comment (laosijikaichele) [reply transport/src/main/java/io/netty/channel/DelegatingChannelPromiseNotifier.java]: @normanmaurer  agree, just deleted comment for this method.

**Comment by normanmaurer** [comment]
@laosijikaichele please sign our ICLA: https://netty.io/s/icla

**Comment by laosijikaichele** [comment]
@normanmaurer  signed done.

**Comment by normanmaurer** [comment]
@Scottmitch fine with you?

**Comment by normanmaurer** [comment]
Squashed and cherry-picked into 4.1 (9dd5c928f3ebf4b3fd8b6cf875d7b170a3a12a48) and 4.0 (16239d4ea3)

**Comment by normanmaurer** [comment]
@laosijikaichele thanks!

**Pull request closed without merging**